package-res/reportviewer/reportviewer.js

```diff
@@ -921,16 +938,92 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
             //Schedule button callback
             function scheduleReport() {
               var urlSchedule = url.substring(0, url.indexOf("/api/repos")) + '/plugin/reporting/api/jobs/' + me._currentReportUuid + '/schedule';
-              pentahoGet(urlSchedule, "");
+              pentahoGet(urlSchedule, "confirm=" + !me.reportPrompt._promptForLocation);
               manuallyScheduled = true;
-              hideDlgAndPane(registry.byId('feedbackScreen'));
 
+              registry.byId('feedbackScreen').hide();
               if (me.reportPrompt._promptForLocation) {
-                //Open location dialog here - see the example of location update below
-             /*   var urlUpdateLocation = url.substring(0, url.indexOf("/api/repos")) + '/plugin/reporting/api/jobs/' + me._currentReportUuid + "/schedule/location?folderId=5a8ba6eb-3d98-4a5c-9357-4a17848925a9&newName=hello_there";
-                pentahoPost(urlUpdateLocation);*/
-              }              
-              
+
+                if(window.top.mantle_addHandler) {
```

Comment:
UpdateReportContentCore is already a huge monster of a function. Is there a chance to refactor this new block so that it is smaller and easier to read? Taking the block out into a new function would be a sweet start, and each of the functions look as if they can be standalone functions (and thus wont pollute the reading flow).

Even the json-parsing parts can be simplified by refactoring it (and the status check) into a function like this:

    (bool) 
    function isStatus (result, expectedStatus) {
      // happy path only, try catch just returns false then ...
      var status = JSON.Parse(result).status 
      var args = Array.prototype.slice.call(arguments, 1);
      return args.includes(expectedStatus);
    }

This cuts out a lot of the ugly clutter that JavaScript forces you to create for such a simple task.